app/invocation/invocation_action_card.tsx

```diff
@@ -335,41 +335,61 @@ export default class InvocationActionCardComponent extends React.Component<Props
     if (filteredEvents.length == 0) {
       return null;
     }
+    const startTimestamp = filteredEvents[0].timestamp;
 
     const totalDuration = durationSeconds(
       filteredEvents[0].timestamp!,
       filteredEvents[filteredEvents.length - 1].timestamp!
     );
 
+    let offset = 0;
     return (
       <div>
+        <div className="metadata-detail">
+          <span className="label">
+            Total
+            {startTimestamp && <> @ {format.formatTimestamp(startTimestamp)}</>}
+          </span>
+          <span className="bar-description">{format.durationSec(totalDuration)} (100%)</span>
+        </div>
+        <div className="action-timeline">
+          <div
+            className="timeline-event"
+            title={`Total: (${format.durationSec(totalDuration)}, 100%)`}
+            style={{ flex: `1 0 0`, backgroundColor: "green" }}></div>
+        </div>
         {filteredEvents.map((event, i) => {
           // Don't render the end marker.
           if (i == filteredEvents.length - 1) return null;
 
           const next = filteredEvents[i + 1];
           const duration = durationSeconds(event.timestamp!, next.timestamp!);
           const weight = duration / totalDuration;
+          offset += weight;
```

Comment:
sorry I realized I forgot to double-check this math :face_with_spiral_eyes: should the `offset += weight` be happening after we render the element? seems to me like the rough pseudo-code should be

```
weight = duration / totalDuration;
bar1Weight = offset;
bar2Weight = weight;
bar3Weight = 1 - weight - offset;
// Offset should apply to *next* bar that is rendered
offset += weight;
// now render bars with bar1Weight, bar2Weight, bar3Weight
```

Reply:
Well spot, I thought of adding it after as well, but that would create a larger diff.

So right now I simply add `weight` into `offset` before the loop. Then inside the loop, I subtract out the `weight` value from each bar, then simplified the calculation. Which resulted in the same value.

This is what it currently looks like on my end 
![image](https://github.com/buildbuddy-io/buildbuddy/assets/26684313/29728b94-2262-4580-885a-400e83799300)